MOV.ino

```diff
@@ -0,0 +1,1077 @@
+      pixel_count--;
+
+    }
+
+    font_col++;
+
+    if (font_col==FONT_WIDTH) {       // Finished with this letter?
+
+      // Move on to next letter
+
+      font_col=0;       
+      
+      s++;
+      len--;
+
+    }
+
+  }
+       
+
+  // If the display is longer than the string, fill any remaining pixels with blanks
+
+  while (pixel_count--) {
+    sendCol(  0  );    // All pixels in column off               
+  }
+    
+  // Latch everything we just sent into the pixels so it is actually displayed
+  show();
+
+  return len>0;     // If len>0 then we ran out of pixels before we ran out of message to display
+}
+
+
+
+
+void setup() {
+
+  Serial.begin(9600);         // Set up the serial port for recieving
+
+  UCSR0B &= ~ _BV( TXEN0 );   // Disable the serial TX pin (digital pin 1) so we can use it as digital IO for an LED string
+                              // Note that you can remove this line if you want to have bidirectional serial communication, 
+                              // but then you will loose a string of LEDS and will need to make the font be only 6 pixels tall.
+                              
+  // initialize LED strip pins
+  
+  PIXEL_DDR |= 0xff;         // Set all 8 pins to output mode (note that digital pin 0 is controlled by the serial port so this has no effect on it)
+
+  // This blinking signal seems to reset some stuck pixels and leaves all outputs cleanly low
+  PIXEL_PORT |= 0xff;       // Set all outputs to 1
+  delay( 100);
+  PIXEL_PORT &= ~0xff;      // Set all outputs to 0
+  delay( 100);
+  PIXEL_PORT |= 0xff;       // Set all outputs to 1
+  delay( 100);
+  PIXEL_PORT &= ~0xff;      // Set all outputs to 0
+  delay( 100);
+
+
+  cli();
+
+ // Show something on startup so we know it is working
+ 
+  appendStringToBuffer( "  .....   booting up  .....    " );
+
+}
+
+
+// Shift keeps track of how many columns shifted over the display currently is. By shifting one column at a 
+// time, we can smoothly scroll text across the display.
+byte shift = 0;
+
+
+
+void loop() {
+
+  
+  for( unsigned int i=0;i < FRAME_DELAY_MS ; i++ ) {
+     _delay_us(900);   // Pool just often enough that we do not miss any serial bytes at 9600bd (1 byte about 1ms) 
+    
+
+    appendStringToBuffer( "...looping...text" );
+    
+  }
+
+
+  // Send out the signals to the LED
+
+  byte more_flag = updateLEDs( buffer , buffer_len , shift );
+
+  if (more_flag) {
+
+    // If there is more of the buffer to display, then we shift over one column before the next update to smooth scroll forward
+
+    shift++;
+
+    // If shifting gets us to the end of the current char, then we move everything forward by one char
+
+    if (shift== FONT_WIDTH) {
+
+      shift = 0 ;
+
+      buffer_len--;
+
+    
+      
+      for( unsigned int i =0; i < buffer_len ; i++ ) {
+          
+          buffer[i] = buffer[i+1];
```

Comment:
so it looks like it's overwriting the buffer, but since you're appending the string so many times at the beginning its growing by 15 every time this loop gets run, which from the looks of it is everytime the pixels advance by one. 

So what I think you want is to do a check at the end and if the buffer_len is 0 you want to reset `buffer_len` and the `buffer` to the original state.